### ISSUE TITLE
Failure to substraction

### ISSUE THREAD

**Issue by QuimMoya** [elalish/manifold#308]
Hello
It seems that at some point boolean substraction fails. I think the input geometry is right. Not sure, but maybe elalish is failing, so i decided to open an issue.

![image](https://user-images.githubusercontent.com/12625508/212007813-70ab26c5-9bf2-42cd-8ae4-a7cecfdb7e33.png)

These are the objects:
[build.zip](https://github.com/elalish/manifold/files/10399260/build.zip)
In the zip you have the first geometry, the geometries to substract and the result.

Thanks for your time.

**Comment by pca006132** [comment]
Maybe you can first check the input geometry with the `IsManifold` method?

**Comment by elalish** [comment]
Agreed; this actually looks like valid output for marginal input (meshes that have faces that are coplanar or nearly so). Due to floating-point rounding error, your positions are not really certain. Therefore it's helpful to think of what you would expect the result to look like if your objects were moved a little (`Precision()`) in different directions. In general we consider any of those potential solutions to be valid output. For cases of exact floating-point equality we do a little better, but that's hard to attain especially with externally-created geometry.

**Comment by QuimMoya** [comment]
I see

Maybe the solution could be to scale the negative solid by a minimal factor.

Thank you.

**Comment by QuimMoya** [comment]
Hey @elalish , Just for curiosity, are you going to implement fuzzy booleans in elalish/manifold to solve these rounding errors?

**Comment by elalish** [comment]
No, I won't. This library intentionally avoids any form of tolerance in the Booleans in order to make them topologically reliable. As a user of the library, you'll either need to avoid creating those marginal cases in the first place, or be very careful with your rounding error to ensure the marginal cases are floating-point equal (which Manifold's transformations help with to some degree).
